### PR TITLE
Use official Ruby setup GH action

### PR DESCRIPTION
**Pull request by seshrs**
This is a CI fix (housekeeping).

## Summary

I noticed that the CI results in #8587 mentioned that [`actions/setup-ruby`](github.com/actions/setup-ruby) is no longer maintained and that we should migrate to using [`ruby/setup-ruby`](github.com/ruby/setup-ruby) instead.

This PR updates jobs to use `ruby/setup-ruby`. I'm not super-familiar with Ruby and had to guess which steps were obsolete with the new action. Hopefully, the GH Actions results on this PR will help evaluate whether I got it right! 😅 

~EDIT: Haha that didn't work. Will try debugging later this week.~

### CODE REVIEW

**Comment by seshrs** [comment]
Oh I see some overlap with #8610. Will wait for that to merge first.

File: .github/workflows/ci.yml
Comment (MichaelCurrin) [inline .github/workflows/ci.yml:87]: For this and the other  config, the docs recommend setting env vars at the job level rather than the step level.

https://github.com/ruby/setup-ruby#bundle-config

I don't know enough about the action to know if it makes a difference, but if i am uncertain I prefer to follow the instructions.

Comment (MichaelCurrin) [reply .github/workflows/ci.yml:87]: Oh nevermind. I think it is just the action author doesn't know you can set the values at the step level.

**Review by MichaelCurrin — APPROVED** [review]

**Comment by MichaelCurrin** [comment]
Looks good but I see there are conflicts to fix and this is WIP.

**Review by MichaelCurrin — CHANGES_REQUESTED** [review]
Fix conflicting file.

**Comment by seshrs** [comment]
@MichaelCurrin yes you were right, the bundle env variables had to be declared at the job-level so that other steps use the correct bundle config! Thanks for pointing that out 😃 

Given that the `Third-Party Repository Profiling` workflow completed successfully, I think we have more confidence that the `docs` workflow will also work when this PR is merged with `master`.

**Comment by MichaelCurrin** [comment]
I recommend making a copy of your existing branch and take out the `on` restriction by branch, so you can see this run in your branch on your fork.

Then you can delete that branch later

**Comment by MichaelCurrin** [comment]
You can find your branch marked with passing runs on the Actions tab 

https://github.com/jekyll/jekyll/actions

**Comment by seshrs** [comment]
Hi @MichaelCurrin, sorry I dropped the ball for a couple of weeks 😅 

Is there some further action needed from me to push this PR forward?

**Comment by MichaelCurrin** [comment]
No PR changes are outstanding. This just needs a review and merging by a maintainer - a reviewer with write access

**Review by parkr — APPROVED** [review]
Looks good to me!

I wasn't able to fully validate the bundler-cache option but I'm assuming we'll be able to continue benefitting from the Actions cache.

**Comment by MichaelCurrin** [comment]
I didn't know about the option when I last reviewed this, but I saw the docs and comment on bundler cache flag here

https://github.com/ruby/setup-ruby#single-job


That means that in this PR, we the action already handles Bundle, bundle install and cache. And so the cache action and the gem install steps are unnecessary and should be taken out. To make the workflow lighter. And also to make the build short and not create conflicts potentially by having two caches and two gem install steps.

**Comment by DirtyF** [comment]
@jekyll: merge +dev

**Pull request merged**
